features/363-glossary/glossary.md

```diff
@@ -0,0 +1,48 @@
+Glossary
+========
+
+In this document, we define a list of terms to be used in communication.
+
+Objective
+---------
+
+In various documents, we will use different terms to refer to concepts in Skygear:
+- Documentation
+- Discussions
+- Emails/Announcements
+
+It would be clearer to use widely agreed terms to refer to specific concepts.
+
+Glossory
+--------
+- **Skygear Cluster**: the Skygear services (Gears, Gateway, and Controller), hosted on Kubernetes.
+- **Skygear Controller**: the management interface of Skygear Cluster. *Should not be used in public documents*.
+- **Skygear Portal**: the web management interface of Skygear Cluster.
+- **Skygear App**: a collection of microservices on Skygear Cluster, can omit 'Skygear' prefix if context if clear.
+- **Skygear Gateway**: the proxy which incoming traffic to Gears and Microservices must go through.
+- **Gear**: officially-supported multi-tenant-aware services.
+    - **Auth Gear**
+    - **Asset Gear**
+- **Deployment**: a unit of executable code belong to an app.
+    - **Microservices**
+- **Developer**: the developer of a Skygear App.
+- **User**: the user of a Skygear App.
+- **Secret**: confidential key value pair belonging to an app.
+- **Collaborator**: developer with management access to app.
+- **User Config**: configuration to gears.
+- **skygear.yaml**: configuration to deployment, including web-hooks.
+- **Scaffold**: the action of initializing an empty directory using template.
+- **Scaffolding Template**:
+- **Access Key**:
+    - **API Key**
+    - **Master Key**
+
+
+Misused Terms
+-------------
+These terms should not be used. Instead, clearer terms from above list should
+be used.
+
+**Skygear Platform**: should use Skygear Cluster.
+**Project**: should use Skygear App.
+**Skygear Auth**: should use Auth Gear.
```

Comment:
I think we need to consider the following: We will describe the major features of Skygear for new comers, especially on marketing / landing pages etc.

In those cases, we will probably want to use the following terms:
* Skygear Microservices
* Skygear Auth
* Skygear Assets

and maybe in future...
* Skygear Cloud Functions
* Skygear Add-ons (e.g. An integrated Helm3 to allow quick and simple deployment of PGSQL, Redis, etc)

I think it would conflict with some terms suggested here. We would probably not want to call them just "Auth Gear" in marketing. Instead, we will sometimes just refer to "Auth, Microservices" etc, or when branding is needed, we call it "Skygear Auth, Skygear Microservices".

We can either:
1. Just use those marketing branded terms in documentations and other places, adopt as standard terms, or...
2. Treat those started with "Skygear X" a categorization of features.
3. Call them "Microservices Gear, Auth Gear, Asset Gear" and "Skygear Microservices Gear, Skygear Auth Gear", etc... but seems a bit too long, and "Microservices Gear" is not exactly correct in technical definition.

Thoughts?